File: 3189-find-champion-ii/3189-find-champion-ii.c
int findChampion(int n, int** edges, int edgesSize, int* edgesColSize) {
    int inDeg[n] = {};
    for (int i = 0; i < edgesSize; ++i) {
        int v = edges[i][1];
        inDeg[v]++;
    }
    int ans = -1, cnt = 0;
    for (int i = 0; i < n; ++i) {
        if (inDeg[i] == 0) {
            ++cnt;
            ans = i;
        }
    }
    return cnt == 1 ? ans : -1;
}

File: 3189-find-champion-ii/3189-find-champion-ii.go
func findChampion(n int, edges [][]int) int {
    inDeg := make([]int, n)
    for i := 0; i < len(edges); i++ {
        v := edges[i][1]
        inDeg[v]++
    }

    cnt, ans := 0, -1
    for i := 0; i < n; i++ {
        if inDeg[i] == 0 {
            cnt++
            ans = i
        }
    }
    if cnt == 1 {
        return ans
    }
    return -1
}